src/care/ProfilePlugin.cpp

```diff
@@ -0,0 +1,54 @@
+#include "care/ProfilePlugin.h"
+#include "care/PluginData.h"
+
+/* CUDA profiling macros */
+#if defined(__CUDACC__) && CARE_HAVE_NVTOOLSEXT
+#include "nvToolsExt.h"
+#endif
+
+namespace care{
+   unsigned int ProfilePlugin::s_current_color = 0;
+   uint32_t ProfilePlugin::s_colors[7] = { 0x0000ff00, 0x000000ff, 0x00ffff00, 0x00ff00ff, 0x0000ffff, 0x00ff0000, 0x00ffffff };
+   int ProfilePlugin::s_num_colors = sizeof(s_colors) / sizeof(uint32_t);
+   bool ProfilePlugin::s_profile_host_loops = true; 
+   
+   void ProfilePlugin::registerPlugin() {
+      static RAJA::util::PluginRegistry::add<care::ProfilePlugin> L ("Profile plugin", "CARE plugin for profiling");
+   }
+
+   void ProfilePlugin::preLaunch(const RAJA::util::PluginContext& p) {
+#if defined(__CUDACC__) && CARE_HAVE_NVTOOLSEXT
```

Comment:
Will something different be needed for HIP ? Maybe add a TODO ?

Reply:
The Hip interface is similar, but we may want to switch to using caliper or even let the host code provide callbacks for profiling. So for now I would just add a comment that profiling is only supported in CUDA builds.

Reply:
That sounds fine. Would you say just a comment is appropriate, or would you suggest a warning or log message to the user, or even a failure/exit?

Reply:
We could output a warning in the registerPlugin function if `__HIPCC__` is defined. I don't think we want a failure/exit.

Reply:
Here's an example of a CARE warning message: https://github.com/LLNL/CARE/blob/8a08fb49ce3ef93837a14595daf4f18a0584eecf/src/care/scan_impl.h#L49